tests/regression/lp-1803542/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Regression test for https://bugs.launchpad.net/snapd/+bug/1803542
+environment:
+    VARIANT/absent: absent
+    VARIANT/present: present
+    VARIANT/shared: shared
+    VARIANT/private: private
```

Comment:
Nice

Reply:
Yeah, worked better than the unit test :)